pkg/scale/decode.go

```diff
@@ -116,51 +111,100 @@ type decodeState struct {
 	io.Reader
 }
 
+//nolint:gocyclo
 func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
 	in := dstv.Interface()
 	switch in.(type) {
 	case *big.Int:
 		err = ds.decodeBigInt(dstv)
+		if err != nil {
+			return fmt.Errorf("decoding big integer: %w", err)
+		}
 	case *Uint128:
 		err = ds.decodeUint128(dstv)
+		if err != nil {
+			return fmt.Errorf("decoding uint128: %w", err)
+		}
 	case int, uint:
 		err = ds.decodeUint(dstv)
+		if err != nil {
+			return fmt.Errorf("decoding uint: %w", err)
+		}
 	case int8, uint8, int16, uint16, int32, uint32, int64, uint64:
 		err = ds.decodeFixedWidthInt(dstv)
+		if err != nil {
+			return fmt.Errorf("decoding fixed width int: %w", err)
```

Comment:
If we are going to prefix the error, I think it needs to print the correct type.